framework/src/modules/interoperability/mainchain/store.ts

```diff
@@ -12,9 +12,137 @@
  * Removal or modification of this copyright notice is prohibited.
  */
 
-//  TODO: extends BaseInteroperabilityStore
-// eslint-disable-next-line  @typescript-eslint/no-extraneous-class
-export class MainchainInteroperabilityStore {
-	// public async forward(ccu: CCUpdateParams, ccm: CCMsg): Promise<void> {};
-	// public async bounce(ccu: CCUpdateParams, ccm: CCMsg): Promise<void> {};
+/*
+ * Copyright © 2022 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+import { BaseInteroperabilityStore } from '../base_interoperability_store';
+import {
+	LIVENESS_LIMIT,
+	MODULE_ID_INTEROPERABILITY,
+	STORE_PREFIX_CHAIN_DATA,
+	STORE_PREFIX_TERMINATED_STATE,
+} from '../constants';
+import {
+	CCMsg,
+	CCUpdateParams,
+	ChainAccount,
+	SendInternalContext,
+	TerminatedStateAccount,
+} from '../types';
+import { chainAccountSchema, terminatedStateSchema } from '../schema';
+
+export class MainchainInteroperabilityStore extends BaseInteroperabilityStore {
+	public async isLive(chainID: Buffer, timestamp: number): Promise<boolean> {
+		const isTerminated = await this.getTerminatedStateAccount(chainID);
+		if (isTerminated) {
+			return false;
+		}
+
+		if (
+			timestamp - (await this.getChainAccount(chainID)).lastCertificate.timestamp >
+			LIVENESS_LIMIT
+		) {
+			return false;
+		}
+
+		return true;
+	}
+
+	public async getChainAccount(chainID: Buffer): Promise<ChainAccount> {
+		const chainSubstore = this.getStore(MODULE_ID_INTEROPERABILITY, STORE_PREFIX_CHAIN_DATA);
+		return chainSubstore.getWithSchema<ChainAccount>(chainID, chainAccountSchema);
+	}
+
+	public async getTerminatedStateAccount(chainID: Buffer): Promise<TerminatedStateAccount> {
+		const terminatedChainSubstore = this.getStore(
+			MODULE_ID_INTEROPERABILITY,
+			STORE_PREFIX_TERMINATED_STATE,
+		);
+		return terminatedChainSubstore.getWithSchema<TerminatedStateAccount>(
+			chainID,
+			terminatedStateSchema,
+		);
+	}
```

Comment:
I think it could be implemented in `BaseInteroperabilityStore` as the implementation will remain the same

Reply:
which function(s) are you referring to precisely ?

Reply:
All the functions implemented here are necessary since `MainchainInteroperabilityStore extends BaseInteroperabilityStore` and all the abstract methods(defined in the previous PR) needs to be implemented otherwise ESlint throws `Non-abstract class 'SidechainInteroperabilityStore' does not implement inherited abstract member`.